Add localStorage

Deardrops opened the pull request.

Deardrops committed:
add localStorage

webxoss commented on src/app.vue:
```diff
@@ -2,20 +2,29 @@
+    // init Deck from localStorage
+    let deck_filenames = localStorage.getItem('deck_filenames')
+    if (deck_filenames) {
+      let FileNames = JSON.parse(deck_filenames)
```
大写！要说几遍？

webxoss commented on src/app.vue:
```diff
@@ -2,20 +2,29 @@
-    })
-    this.$store.commit('switchDeck', 'WHITE_HOPE')
+    // init Deck from localStorage
+    let deck_filenames = localStorage.getItem('deck_filenames')
```
js 变量名用驼峰

webxoss commented on src/app.vue:
```diff
@@ -2,20 +2,29 @@
+        }
+      })
+      this.$store.commit('switchDeck', FileNames[0])
+    } else {
```
如果`deck_filenames `长度为0呢？
这里不要 else ，直接 if 没有卡组

webxoss commented on src/store.js:
```diff
@@ -150,4 +150,19 @@ const store = new Vuex.Store({
+
+// watch deck change && save to localStorage immediately
+store.watch((state, getters) => {
+  if (getters.deckNames.length === 0) return
```
if 都不要单行吧

webxoss commented on src/store.js:
```diff
@@ -150,4 +150,19 @@ const store = new Vuex.Store({
+    mainDeck: getters.mainDeck.map(card => card.pid),
+    lrigDeck: getters.lrigDeck.map(card => card.pid),
+  }
+  if (state.deckName.length === 0) return
```
if 都不要单行吧

Deardrops committed:
fix review

camel-case variable
check deck_filenames is null or its length is 0
"if" with newlines

webxoss commented on src/app.vue:
```diff
@@ -2,20 +2,29 @@
+      return
+    }
+    fileNames.forEach(name => {
+      let deckFile = localStorage.getItem('deck_file_'+name)
```
操作符前后空格！
这里应该用ES6模板

webxoss commented on src/app.vue:
```diff
@@ -2,20 +2,29 @@
+    // init Deck from localStorage
+    let deckFileNames = localStorage.getItem('deck_filenames')
+    let fileNames = JSON.parse(deckFileNames)
+    if (!fileNames || fileNames.length === 0) {
```
不要`=== 0`直接取反，下次再这样扣工资

webxoss commented on src/store.js:
```diff
@@ -150,4 +150,21 @@ const store = new Vuex.Store({
+
+// watch deck change && save to localStorage immediately
+store.watch((state, getters) => {
+  if (!getters.deckNames || getters.deckNames.length === 0)
```
同上

webxoss commented on src/store.js:
```diff
@@ -150,4 +150,21 @@ const store = new Vuex.Store({
+// watch deck change && save to localStorage immediately
+store.watch((state, getters) => {
+  if (!getters.deckNames || getters.deckNames.length === 0)
+    return
```
加上花括号。。。这样太难看了

webxoss commented on src/store.js:
```diff
@@ -150,4 +150,21 @@ const store = new Vuex.Store({
+    lrigDeck: getters.lrigDeck.map(card => card.pid),
+  }
+  if (!state.deckName || state.deckName.length === 0)
+    return
```
同上

webxoss commented on src/store.js:
```diff
@@ -150,4 +150,21 @@ const store = new Vuex.Store({
+    mainDeck: getters.mainDeck.map(card => card.pid),
+    lrigDeck: getters.lrigDeck.map(card => card.pid),
+  }
+  if (!state.deckName || state.deckName.length === 0)
```
同上

webxoss reviewed (commented):
都是review可以考虑amend
